### ISSUE TITLE
Remove "api_endpoint" from the configuration settings?

### ISSUE THREAD

**Issue by ttmc** [bigchaindb/bigchaindb#780]
The `api_endpoint` is [a configuration setting](https://docs.bigchaindb.com/projects/server/en/latest/server-reference/configuration.html#api-endpoint).

I think it was originally going to be used by the `bigchaindb load` command (for load testing), but I don't think it's being used for that. It appears that command uses the `database.host`, `database.port` and `database.name` config settings to know where to write the transactions.

As far as I can tell, the `api_endpoint` setting isn't being used anywhere in the functional part of the code. For example, it's not used by the HTTP API web server. The URL prefix "/api/v1" for the transaction endpoints [is actually hard-coded](https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/web/server.py#L68) (and that's okay).

To do:

- [x] Remove the `api_endpoint` setting, and all the code associated with it. Don't forget the AWS deployment scripts.
- [x] Update the docs about the BigchainDB Server Configuration Settings: remove the section about `api_endpoint`. Write something about how the API URL is actually determined.

**Comment by ttmc** [comment]
What _does_ determine the actual API endpoint URL? Examples:

http://12.34.56.78:9984/api/v1/

http://lovelyday.net:9984/api/v1/

## Answer

This is how I understand it (but I may be wrong):

* The public IP address (like 12.34.56.78) is the public IP address of the machine exposing the HTTP API to the public internet (e.g. either the machine hosting Gunicorn or the machine running the reverse proxy such as Nginx). It's determined by AWS, Azure, Rackspace, or whoever is hosting the machine.

* The DNS hostname (like lovelyday.net) is determined by DNS records, such as an "A Record" associating lovelyday.net with 12.34.56.78

* The port (like 9984) is determined by the `server.bind` setting if Gunicorn is exposed directly to the public Internet. If a reverse proxy (like Nginx) is exposed directly to the public Internet instead, I suppose it could expose the HTTP API on whatever port it wants to. (It _should_ expose the HTTP API on port 9984, but it's not bound to do that by anything other than convention. It certainly doesn't care, or even know, about the value of the `api_endpoint` setting. That's a BigchainDB Server setting, not an Nginx setting.)

* `/api/v1/` is [hard-coded in the Flask code for the transaction_views](https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/web/server.py#L68).

* `/` is hard-coded in the Flask code for the info_views.

If all of the above is correct, then none of the HTTP API endpoint URL comes from the `api_endpoint` setting.

**Comment by vrde** [comment]
Originally `api_endpoint` was used to configure the client. Now that the driver is separated from the server, so it is no longer in use.

If I grep for `api_endpoint`, this is what I see from `master`:
```
 ~/a/r/bigchaindb git:(master) ➜ ag api_endpoint bigchaindb/
bigchaindb/__init__.py
32:    'api_endpoint': os.environ.get('BIGCHAINDB_API_ENDPOINT') or 'http://localhost:9984/api/v1',

bigchaindb/web/views/info.py
24:        'api_endpoint': bigchaindb.config['api_endpoint']
```

TL;DR: It's safe to remove it.

**Comment by ttmc** [comment]
@sbellem In the root directory of bigchaindb/bigchaindb, the `Dockerfile` contains the line

```text
ENV BIGCHAINDB_API_ENDPOINT http://bigchaindb:9984/api/v1
```

I guess [that sets the environment variable](https://docs.docker.com/engine/reference/builder/#/env) `BIGCHAINDB_API_ENDPOINT` to `http://bigchaindb:9984/api/v1`. We now know that BigchainDB Server  doesn't need or use the value of that environment variable, but does _something else_ use it?

There's a line in `docker-compose.yml`:

```text
BIGCHAINDB_API_ENDPOINT: http://bdb:9984/api/v1
```

What does that do?

My thinking is that I'll remove or comment-out all the `api_endpoint` stuff but that may break some Docker things and I may need your help to get them working again.
